action.yml

```diff
@@ -0,0 +1,28 @@
+---
+author: Cybersecurity and Infrastructure Security Agency
+branding:
+  color: blue
+  icon: shield
+description: >-
+  Our standard permissions monitoring and runner hardening that is to
+  be applied at the beginning of each GitHub Actions job.
+name: Job preamble
+
+inputs:
+  actions_permissions_config:
+    description: >-
+      The permission configuration to use.  You will usually want to
+      set this to `${{ vars.ACTIONS_PERMISSIONS_CONFIG }}`.
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: GitHubSecurityLab/actions-permissions/monitor@v1
+      with:
+        config: ${{ inputs.actions_permissions_config }}
+    - id: harden-runner
+      name: Harden the runner
+      uses: step-security/harden-runner@v2
+      with:
+        egress-policy: audit
```

Comment:
I think this should also be an input and we can configure it to have `audit` as a default value if preferred.

Reply:
Please see commit 9d98200.